bzip2.rs

```diff
@@ -1627,26 +1674,24 @@ unsafe fn compress(name: *mut c_char) {
                 b"(stdout)\0" as *const u8 as *const libc::c_char,
             );
         }
-        3 => {
+        SourceMode::F2O => {
             copyFileName(inName.as_mut_ptr(), name);
-            copyFileName(outName.as_mut_ptr(), name);
-            strcat(
+            copyFileName(
                 outName.as_mut_ptr(),
-                b".bz2\0" as *const u8 as *const libc::c_char,
+                b"(stdout)\0" as *const u8 as *const libc::c_char,
             );
         }
-        2 => {
+        SourceMode::F2F => {
             copyFileName(inName.as_mut_ptr(), name);
-            copyFileName(
+            copyFileName(outName.as_mut_ptr(), name);
+            strcat(
                 outName.as_mut_ptr(),
-                b"(stdout)\0" as *const u8 as *const libc::c_char,
+                b".bz2\0" as *const u8 as *const libc::c_char,
```

Comment:
note: the branches get flipped around here so that they are in the logical order (according to the enum). but that does mean the diff is kinda weird.